src/store/reducers/userReducers.ts

```diff
@@ -37,6 +36,12 @@ export const logoutRequest = createAsyncThunk('user/logoutRequest', async () =>
   await api.logout();
 });
 
+export const userRequest = createAsyncThunk('user/userRequest', async () => {
+  const response = await api.getUser();
+
+  return response;
```

Comment:
❗️
Такое безболезненно можно сокращать
```suggestion
  return await api.getUser();
```

Reply:
Вот [тут](https://jakearchibald.com/2017/await-vs-return-vs-return-await/) есть о том, когда надо, а когда не стоит)

Reply:
Идентичны, и следуя правилу как бы "неправильны", т.е. надо писать `return api.getUser()`. Во многих случаях (у нас во всём коде подобные строчки/случаи) это будет вроде как нормально — если мы не обрабатываем ошибки, то возвращаем такой же промис, как и через await, просто он разрешится самостоятельно, а мы его в функции ждать не будем. Исключение когда мы деструктурируем результат `const { val } = await api.method()`, тут уж точно нужен `await`. 

Но здесь у нас редьюсер, т.е. может ожидается какой-то особый возврат, хотя с `await` мы тот же промис возвращаем, просто который уже точно зарезолвится. Мне кажется `return api.getUser()` здесь будет работать?

Reply:
С другой стороны зачем нам тогда `async` функция, если мы не используем `await`. Она неявно возвращает промис, не знаю хорошо ли это. Ну хотя бы для консистентности, т.к. где-то `await` используется. А то что она возвращает промис просто надо уяснить, хотя всё равно как то странно, не зря же речь зачастую идёт про `async/await`, а не просто про `async`

Reply:
И зачем тогда `thunk`, если мы не используем `async`-экшены)
Возвращать исходный промис можно, если ты потом хочешь его где-то обрабатывать. Тут же у тебя `createAsyncThunk` сделает это за тебя, порулив всё, что нужно под капотом. Если посмотреть на их [доку](https://redux-toolkit.js.org/api/createAsyncThunk), то они советуют писать так же и это ок в данном кейсе

Reply:
https://github.com/eslint/eslint/issues/14542

Reply:
https://github.com/eslint/eslint/issues/12246#issuecomment-535737777
https://github.com/standard/standard/issues/1442#issuecomment-592150014
https://github.com/standard/standard/issues/1442#issuecomment-592318304
https://github.com/standard/standard/issues/1442#issuecomment-592400623

Вобщем я согласен с тем, что формально `await` не нужен, т.е. надо писать `async` функцию и возвращать вызов api без `await`. Стэк-трэйсы это проблема движка (но нас они вряд ли вообще будут интересовать), производительность — второстепенный вопрос.

В целом выглядит логично - использовать `await`, когда действительно нужно ждать промис в функции (обработка ошибок, деструктуризация полей). В ином случае он не имеет смысла, т.к. `async`-функция уже возвращает свой промис, который мы и так будем ждать (и возможно `async` функция даже сама ждёт резолва возвращаемого промиса, перед тем как передать результат в свой возвращаемый промис, но не уверен, надо спецификацию читать: https://github.com/standard/standard/issues/1442#issuecomment-594052152)

Альтернативное мнение: https://github.com/standard/standard/issues/1442#issuecomment-592404483

P.S.: " зачем нам тогда async функция" - затем чтобы возвращать промис, т.к. действие асинхронное (async thunk). Как понимаю, async thunk можно написать c обычными промисами, просто будет больше кода. Короче async-функция это более простой способ использовать промисы

Reply:
Насчёт доков `createAsyncThunk`, там почти везде `await` оправдан, т.е. они либо ошибки обрабатывают, либо с ответом что-то делают, в отличии от нашего кода, где просто возвращается вызов api

Reply:
Сама асинхронная функция ничего не ждёт, с ней просто работаешь, как с промисом, не путай всё в одну кучу)
Насчёт возвращения промиса из `api` в целом довольно логично, но в таком случае давай везде сделаем консистентно

Reply:
Ну я уже не уверен что не ждёт, не удивлюсь что и ждёт после всего что прочитал, хотя понимаю что скорей всего нет. Буду потом в спецификации смотреть

Reply:
Этот момент надо отражать в кодстайле, он довольно неоднозначный. Использовать await только для реальной необходимости. Использовать async если нужно вернуть промис. Даже сложно сразу внятно сформулировать, о чём писать в кодстайле

Reply:
Хотя сами слова понятные: await — ждёт, async — возвращает промис

Reply:
Хотя... Это всё как раз это правило и регулирует — `no-return-await`, может и писать ничего не надо. Просто указать что код как выше писать не надо (ждать и присваивать в переменную а потом её же возвращать)